include/random_walks/uniform_accelerated_billiard_walk.hpp

```diff
@@ -13,6 +13,112 @@
 
 #include "sampling/sphere.hpp"
 #include <Eigen/Eigen>
+#include <set>
+#include <vector>
+
+const double eps = 0.000000001;
+
+template<typename NT>
+class Heap {
```

Comment:
Could you please add a few lines of comments to briefly explain what does this data structure does?
You could add what is the improvement it gives regarding the previous implementation.
Don't be afraid to add some equations to help your explanation, e.g., how Av and Ar are updated now etc.